### PR TITLE
Support QML and Web content in overlay windows

### PR DESCRIPTION
**Pull request by jherico**
90% of the work required to get arbitrary user QML content in overlays from scripts was already present thanks to the HTML window support.  This updates the code to support QML as well.  

Todo:

* support the event bridge, similar to our support in HTML
* support our existing script interfaces directly

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: ecff1b3ef60183dda94b09f2821e8cb5dc8167f1 available at: https://gist.github.com/31102cb372e020fa5a2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6739.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6739.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e01083b42054de31039fa28d732d83f0cc131c04 available at: https://gist.github.com/31102cb372e020fa5a2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6739.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6739.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3e91977836ae7350c5ac94432c140234740876ba available at: https://gist.github.com/31102cb372e020fa5a2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6739.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6739.dmg

**Comment by imgntn** [comment]
@jherico tested on win10 with the test script and seems to be working well :+1:

File: examples/tests/qmlWebTest.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 print("Launching web window");
 
-webWindow = new OverlayWebWindow('Test Event Bridge', "file:///C:/Users/bdavis/Git/hifi/examples/html/qmlWebTest.html", 320, 240, false);
+var htmlUrl =  Script.resolvePath("..//html/qmlWebTest.html")
```
Comment (imgntn) [inline examples/tests/qmlWebTest.js]: should this have one less slash ```Script.resolvePath("../html/qmlWebTest.html")```

**Comment by hifi-gustavo** [comment]
Build console output for commit: 31cb6972a5cad5a02fd36c794a49ec090827de09 available at: https://gist.github.com/31102cb372e020fa5a2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6739.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6739.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d6f5296c3b68d265e801128ccdb918401cccb763 available at: https://gist.github.com/31102cb372e020fa5a2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6739.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6739.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f4bd2afc8e4dd5417821babfdc310d9ec5ac7dd1 available at: https://gist.github.com/31102cb372e020fa5a2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6739.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6739.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bfe7ab4d9431c53d67a62fa3963d7c209a593cb8 available at: https://gist.github.com/31102cb372e020fa5a2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6739.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6739.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4c26627622afdea02276556ee7fc61c505f60e6b available at: https://gist.github.com/31102cb372e020fa5a2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6739.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6739.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a883891197c95092cbefc48bb145c89b24e79f23 available at: https://gist.github.com/31102cb372e020fa5a2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6739.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6739.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 462cc325e5b03c46b53360c821e0bef3418e8c87 available at: https://gist.github.com/31102cb372e020fa5a2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6739.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6739.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 67e32cc1f0e9dd6afd949510185714fe724691d3 available at: https://gist.github.com/31102cb372e020fa5a2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6739.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6739.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a0196f0cde408f20cfd1e6c4e58dc5ac2165d485 available at: https://gist.github.com/31102cb372e020fa5a2b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6739.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6739.dmg

File: libraries/controllers/src/controllers/UserInputMapper.cpp [hunk 2]
```diff
@@ -888,7 +903,7 @@ Endpoint::Pointer UserInputMapper::parseDestination(const QJsonValue& value) {
 
 Endpoint::Pointer UserInputMapper::parseAxis(const QJsonValue& value) {
     if (value.isObject()) {
-        auto object = value.toObject();
+        auto object = value.toObject();
```
Comment (ZappoMan) [inline libraries/controllers/src/controllers/UserInputMapper.cpp]: extra whitespace - not a merge blocker

**Pull request merged**